components/brave_ads/core/internal/account/confirmations/queue/confirmation_queue_unittest.cc

```diff
@@ -297,24 +300,23 @@ TEST_F(BraveAdsConfirmationQueueTest,
          BuildCreateNonRewardConfirmationUrlResponseBodyForTesting()}}}};
   MockUrlResponses(ads_client_mock_, url_responses);
 
-  AdvanceClockTo(
-      TimeFromString("November 18 2023 19:00:00.000", /*is_local*/ true));
-
   const TransactionInfo transaction = BuildUnreconciledTransactionForTesting(
       /*value*/ 0.01, ConfirmationType::kViewed,
       /*should_use_random_uuids*/ false);
 
+  // Act
   {
     const absl::optional<ConfirmationInfo> confirmation =
         BuildNonRewardConfirmation(transaction,
                                    /*user_data*/ {});
     ASSERT_TRUE(confirmation);
 
+    ScopedTimerDelaySetterForTesting scoped_setter(base::Seconds(7));
     confirmation_queue_->Add(*confirmation);
 
     ASSERT_TRUE(confirmation_);
     ASSERT_TRUE(did_add_to_queue_);
-    ASSERT_TRUE(will_process_queue_at_);
+    ASSERT_EQ(Now() + base::Seconds(7), will_process_queue_at_);
```

Comment:
I think this can be flaky because current Now() can differ from one during `will_process_queue_at_` calculation.

Reply:
We do not call `task_environment_.FastForwardBy` or `task_environment_.AdvanceClock` so the clock will not change, unless of course I am missing something?

Reply:
We use `task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME)`